pkg/utils/kubeconfig/kubeconfig.go

```diff
@@ -183,6 +183,12 @@ func AppendAuthenticator(config *clientcmdapi.Config, clusterMeta *api.ClusterMe
 			})
 		}
 	case AWSEKSAuthenticator:
+		// if version is above or equal to v0.5.3 we change the APIVersion to v1beta1.
+		if authenticatorIsBetaVersion, err := authenticatorIsAboveVersion(AWSIAMAuthenticatorMinimumBetaVersion); err != nil {
```

Comment:
This is checking aws authentication version. But didn't you say that it's not installed? That would just error and default back to alphav1. This should check the AWS CLI version and verify which is the minimum version at which it will provide the betav1 version. But very good start! :)

Reply:
Yep, thanks for quick response. I've updated the PR 👍